### DISCUSSION TITLE
How to radar all recyclable objects in CBot

### DISCUSSION THREAD

**Discussion by pestoffne** [colobot/colobot#1417] [Q&A] [answered]
At first glance it seems that you can just `radar(Wreck)` to find recyclable object. But I catch sight of that `radar(Wreck)` finds only some recyclable objects.

For example in `Returbon (Grabbers Edition) 1.2b`
https://www.moddb.com/games/colobot-gold-edition/addons/returbot-grabbers-edition
in RB2 in fourth mission which on the screenshot we can see 7 "destroyed" objects. Only 2 objects with categories 601 and 603 are finds with `radar(Wreck)`, but objects 607, 608 and 609 are recyclable too. Only one object with category 606 on that picture is not recyclable.

![recycle](https://user-images.githubusercontent.com/12718711/117542799-42358300-b044-11eb-91d2-ccd52a23247d.png)

Not a very elegant way to find all recyclable items in this particular case. 

```
int[] recyclable; int i; object item;

i = 0;
recycleble[i++] = Wreck;
recycleble[i++] = 607;
recycleble[i++] = 608;
recycleble[i++] = 609;

for (;;) {
    item = radar(recyclable);
    // do some stuff to recycle item an so on
}
```

**Comment by ghost** [comment, marked as answer]
Broken parts of buildings and wrecks of bots are different types of objects and they have different names, Ruin and Wreck. Or so is said in satcom, because compiler doesn't actually understand Ruin.

It seems that because recycler was never supposed to be able to recycle ruins originally, ruins were separated from wrecks.

They [revert ability to recycle ruins](https://github.com/colobot/colobot/commit/c50b36118efa8f936f36763c64a03445222983e1), therefore it will no be a problem anymore.

**Reply by pestoffne** [reply]
Yes, only wrecks will be recyclable since next release. Thank you for answer.
